src/states/proving_node.rs

```diff
@@ -39,62 +39,71 @@ use std::{
     fmt::{self, Display, Formatter},
 };
 
+pub struct ProvingNodeDetails {
+    pub action_sender: RoutingActionSender,
+    pub cache: Box<Cache>,
+    pub crust_service: Service,
+    pub full_id: FullId,
+    pub min_section_size: usize,
+    pub old_full_id: FullId,
+    pub our_section: (Prefix<XorName>, BTreeSet<PublicId>),
+    pub proxy_pub_id: PublicId,
+    pub timer: Timer,
+}
+
 pub struct ProvingNode {
-    pub(super) ack_mgr: AckManager,
```

Comment:
Wouldn't it make sense to just use composition to avoid repeating all these fields?